File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        //stack method
        Stack<Integer> stack = new Stack<>();
        for(String token : tokens) {
            if(!"+-*/".contains(token)) {
                stack.push(Integer.parseInt(token));
                continue;
            }
            int res = 0;
            int num2 = stack.pop();
            int num1 = stack.pop();
            
            switch(token) {
                case "+":
                    res = num1 + num2;
                    break;
                case "-":
                    res = num1 - num2;
                    break;
                case "*":
                    res = num1 * num2;
                    break;
                case "/":
                    res = num1 / num2;
                    break;
            }
            stack.push(res);
        }
        return stack.pop();
    }
}

File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = collections.deque()
        for s in tokens:
            if s not in '+-*/':
                stack.append(int(s))
            else:
                val1,val2 = stack.pop(), stack.pop()
                if s == '+':
                    stack.append(val2 +val1)
                elif s == '-':
                    stack.append(val2 - val1)
                elif s == '*':
                    stack.append(val1 * val2)
                else:
                    stack.append(int(float(val2) / val1))
        return stack.pop()
        